src/phoenix/datasets/dataset.py

```diff
@@ -121,6 +124,142 @@ def from_name(cls, name: str) -> "Dataset":
         schema = Schema.from_json(schema_json)
         return cls(df, schema, name)
 
+    @classmethod
+    def from_open_inference(cls, dataframe: DataFrame) -> "Dataset":
+        schema = Schema()
+        column_renaming: Dict[str, str] = {}
+        for group_name, group in groupby(
```

Comment:
Suggestion: `column_name` and `column` are clearer and more meaningful names IMO.

Reply:
`group_name` is appropriate because grouping things is its primary function